lib/covenants/rules.js

```diff
@@ -583,23 +606,23 @@ rules.countRenewals = function countRenewals(tx) {
 
   let total = 0;
 
-  for (const {covenant} of tx.outputs) {
```

Comment:
Unless you can prove that your refactor is faster, I do not think that this code should change.